ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -53,11 +54,23 @@ public DDLCommandResult run(MetaStore metaStore) {
           dataSourceType == DataSource.DataSourceType.KSTREAM ? "STREAM" : "TABLE"
       ));
     }
-    DropTopicCommand dropTopicCommand =
-        new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
+    if (!referentialIntegrityTable.isSafeToDrop(sourceName)) {
+      String sourceForQueriesMessage = referentialIntegrityTable
+          .getSourceForQuery(sourceName).stream()
+          .collect(Collectors.joining(", "));
+      String sinkForQueriesMessage = referentialIntegrityTable.getSinkForQuery(sourceName)
+          .stream()
+          .collect(Collectors.joining(", "));
+      throw new KsqlException(String.format("Cannot drop the data source. The following queries "
+                                            + "read from this source: [%s] and the following "
+                                            + "queries write into this source: [%s]. You need to "
+                                            + "terminate them before dropping this source.",
+                                            sourceForQueriesMessage, sinkForQueriesMessage));
+    }
+    DropTopicCommand dropTopicCommand = new DropTopicCommand(
+        dataSource.getKsqlTopic().getTopicName());
     dropTopicCommand.run(metaStore);
     metaStore.deleteSource(sourceName);
-    queryTerminator.terminateQueryForEntity(sourceName);
```

Comment:
Why are we no longer terminating the query when dropping? Or are we now changing the semantics to require the user to terminate the query before dropping the source? (which would answer my above question). If so this merits a doc update.

Reply:
Yes, now user should terminate all the queries that use the stream/table before being able to drop it.